### PR TITLE
Fix #1021: Fixed bug of crash while clicking profile in navigation-bar in landscape

### LINKED ISSUE #1021: Crash on Profile Image click on Navigation Drawer landscape
Go to the home page, set the orientation to landscape and then open the side navigation drawer and click on the profile image.

### PR DESCRIPTION
**Pull request by abhinavraj23**
## Explanation
Fixes #1021 : Fixed bug of crash while clicking profile in navigation-bar in landscape by retaining navigation bar instance

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt [hunk 1]
```diff
@@ -38,6 +38,7 @@ class HomeActivityPresenter @Inject constructor(private val activity: AppCompatA
       activity.findViewById<View>(R.id.home_activity_drawer_layout) as DrawerLayout,
       toolbar, R.id.nav_home
     )
+    navigationDrawerFragment!!.retainInstance = true
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt]: Actually this is not a correct solution for this.
The problem is in `NavigationDrawerFragment` -> `routeToProfileProgress`, when we do orientation change the activity gets destroyed and the lifecycle of fragment is independent of that and thats why `this.activity!!` is creating issues as it is null. To fix this, we should actually call intent in the `NavigationDrawerFragmentPresenter` to fix this. Like this:


Code in `NavigationDrawerFragmentPresenter`

```
fun openProfileProgress(profileId: Int) {
    activity.startActivity(ProfileProgressActivity.createProfileProgressActivityIntent(activity, profileId))
  }
```

Inside `routeToProfileProgress` you can just call this presenter function.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Sugessted changes.

File: app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt (hunk 1 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt]: @rt4914 thanks for the suggestion, I have changed the implementation and it is working fine now!

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**